detection_rules/rule.py

```diff
@@ -30,7 +30,7 @@ def __init__(self, path, contents):
         """Create a Rule from a toml management format."""
         self.path = os.path.abspath(path)
         self.contents = contents.get('rule', contents)
-        self.metadata = self.set_metadata(contents.get('metadata', contents))
+        self.metadata = contents.get('metadata', self.set_metadata(contents))
```

Comment:
i thought `set_metadata` takes the `metadata` dict, not the full rule dict?

Reply:
it was done this way initially to be more lenient if information was in a flat structure. This change should be more resilient now since it will preserve passed metadata that is _not_ in the schema, which would fail as intended. If we want to be more strict, we can drop the fallback to search directly in `contents`, but I _think_ some of the rule importing _may_ have been dependent on it

Reply:
but `set_metadata` wasn't changed. contents will contain two keys: `rule` and `metadata`. but doesn't set_metadata expect the contents _inside_ `metadata`, like `updated_at`, `ecs_versions`, etc?